### PR TITLE
fix(SSG): RHICOMPL-3259 do not cache but memoize SupportedSSg.revision

### PR DESCRIPTION
**Pull request by skateman**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Review by vkrizan — APPROVED** [review]
Make sense.

File: lib/tasks/import_datastream.rake [hunk 1]
```diff
@@ -6,14 +6,18 @@
...
   task check_synced: [:environment] do
-    datastreams_message = "SSG datastreams not synced\nDatastream config revision: " \
-      "#{SupportedSsg.revision.inspect}\nDB revision: " \
-      "#{Revision.datastreams.inspect}"
-    remediations_message = "SSG remediations not synced\nRemediation config revision: " \
-      "#{SupportedRemediations.revision.inspect}\nDB revision: " \
-      "#{Revision.remediations.inspect}"
-    abort datastreams_message if Revision.datastreams != SupportedSsg.revision
-    abort remediations_message if Revision.remediations != SupportedRemediations.revision
+    if Revision.datastreams != SupportedSsg.revision(true)
```
Comment (vkrizan) [inline lib/tasks/import_datastream.rake:9]: The force is very subtle here. If one would remove the first condition the second one would have again cached values. Maybe add a comment?

Comment (skateman) [reply lib/tasks/import_datastream.rake:9]: This is the only place this check is actually needed, if someone removes it, things will go bad on many other levels.

**Pull request merged**